components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/tool_event.tsx

```diff
@@ -0,0 +1,135 @@
+import Tooltip from '@brave/leo/react/tooltip'
+import classnames from '$web-common/classnames'
+import { createTextContentBlock } from '../../../common/content_block'
+import * as Mojom from '../../../common/mojom'
+import ConversationAreaButton from '../../../common/components/conversation_area_button'
+import { useUntrustedConversationContext } from '../../untrusted_conversation_context'
+import styles from './tool_event.module.scss'
+
+interface Props {
+  toolUseEvent: Mojom.ToolUseEvent
+  isEntryActive: boolean
+}
+
+// Content customizable for each known tool type
+interface ToolUseContent {
+  toolText: JSX.Element
+  tooltipContent: JSX.Element | null
+  statusIcon: JSX.Element
+  progressIcon: JSX.Element
+}
+
+function useUserChoiceToolContent(props: Props, toolInput: any): Partial<ToolUseContent> {
+  const context = useUntrustedConversationContext()
+  const content: Partial<ToolUseContent> = {}
+
+  if (props.toolUseEvent.output) {
+    // Tool already completed, don't allow further response
+    content.toolText = (
+      <ConversationAreaButton
+        onClick={() => {}}
+        isDisabled={true}
+        icon={<Icon className={styles.completedChoiceIcon} name='checkbox-checked' />}
+      >
+        {props.toolUseEvent.output[0]?.textContentBlock?.text}
+      </ConversationAreaButton>
+    )
+  } else {
+    content.progressIcon = <Icon name='help-outline' />
+
+    const handleChoice = (choice: string) => {
+      if (!props.isEntryActive) {
+        return
+      }
+      context.conversationHandler?.respondToToolUseRequest(
+        props.toolUseEvent.id, [createTextContentBlock(choice)]
+      )
+    }
+
+    content.toolText = (
+      <>
+        {toolInput?.choices?.map((choice: string, i: number) => (
+          <div key={i} className={styles.toolChoice}>
+            <ConversationAreaButton
+              className={classnames(styles.choice)}
+              isDisabled={!props.isEntryActive}
+              onClick={() => handleChoice(choice)}
+              icon={<div className={classnames(styles.choiceNumber)}>{i+1}</div>}
+            >
+              {choice}
+            </ConversationAreaButton>
+          </div>
+        ))}
+      </>
+    )
+  }
+
+  return content
+}
+
+export function useToolEventContent(props: Props): ToolUseContent {
+  const {toolUseEvent} = props
+
+  let content: ToolUseContent = {
+    toolText: <>{toolUseEvent.toolName}</>,
+    tooltipContent: null,
+    statusIcon: <Icon name="check-circle-outline" />,
+    progressIcon: <ProgressRing />
+  }
+
+  const input = React.useMemo(() => {
+    if (!toolUseEvent?.argumentsJson) {
+      return null
+    }
+    try {
+      return JSON.parse(toolUseEvent.argumentsJson)
+    } catch (e) {
+      return null
+    }
+  }, [toolUseEvent?.argumentsJson])
+
+  if (toolUseEvent.toolName === Mojom.USER_CHOICE_TOOL_NAME) {
+    content = { ...content, ...useUserChoiceToolContent(props, input)}
+  }
```

Comment:
I don't think this is safe - we shouldn't be using hooks in branches.

I think instead `useUserChoiceToolContent` should just be a normal function we can pass what we need from `useUntrustedConversationContext` into.

Reply:
So I ended up making this a component which passes the params to a child function (or render callback or whatever that pattern is called!). I'm not 100% sure that's the best way but the design is ever changing for this so who knows if every event will take on this template or not. It will probably change with the next tool! So for now I guess this new pattern is as good as any as long as it fixes the conditional hook call (which I think it does)